src/api/app/models/bs_request.rb

```diff
@@ -623,7 +624,7 @@ def change_review_state(new_review_state, opts = {})
       found = false
 
       reviews_seen = Hash.new
-      reviews.reverse_each do |review|
+      reviews.lock(true).reverse_each do |review|
```

Comment:
By default `lock` is `true`, http://apidock.com/rails/ActiveRecord/QueryMethods/lock